worker/orca_grader/docker_utils/images/image_loading.py

```diff
@@ -25,10 +26,15 @@ def load_image_from_tgz(tgz_file_path: str) -> None:
     ]
     _LOGGER.debug(f"Attempting to load image from file {tgz_file_path}")
     result = subprocess.run(program_args, check=True, capture_output=True)
-    _LOGGER.debug(
-        f"Docker save stdout: {result.stdout.decode() if result.stdout is not None else '<None>' }")
-    _LOGGER.debug(
-        f"Docker save stderr: {result.stderr.decode() if result.stderr is not None else '<None>' }")
+    _LOGGER.debug(f"Docker save stdout: {result.stdout.decode() if result.stdout is not None else '<None>' }")
+    _LOGGER.debug(f"Docker save stderr: {result.stderr.decode() if result.stderr is not None else '<None>' }")
     _LOGGER.debug("Image loaded.")
     # os.remove(tgz_file_path)  # To save resources, clean up tgz after load.
     # _LOGGER.debug("Image tgz file removed.")
+    return result.stderr and get_name_from_load_output(result.stderr.decode())
+
+
+def get_name_from_load_output(stderr: str) -> Optional[str]:
+    pattern = re.compile(r"^Loaded image\: (.*)$")
```

Comment:
I'm surprised you have to escape the colon here, and I'm also surprised this is robust enough in general -- do you know if stderr contains multiple lines?  (If so this might not match.)  Do you mean stderr or stdout?  Would it be better to match against `r^"Loaded image: ([a-zA-Z0-9/:_-]+)"` instead?

Reply:
Docker writes basically all of its output from running commands to `stderr` (unless you are explicitly running a command to get information), confirmed in this scenario as well. That regex certainly seems more reasonable

Will double check on Python regex with escaping colons

Reply:
Scratch that, it was `stdout` in this case